### PR TITLE
Redirect with message when root category is not available for a shop

### PR DESCRIPTION
**Pull request by aleeks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Redirect with message when root category is not available for a shop
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | only @dSkrbic knows. See after: 

1/ Activate multishop
2/ Create a new category root
3/ Create a shop using this new category root 
4/ Edit the new category, unselect the new shop (which used this category root)
5/ Go to your dashboard, select the new shop 
6/ try to access to the category page. You should be redirected to the dashboard with a message error
Before, you should have a 404

Sorry for this code so 2000', apply same system of &conf for alert-danger.... :sob::sob::sob:

### CODE REVIEW

**Comment by aleeks** [comment]
ping @AlexEven  for wording!!!!!!!!! (domain)

File: classes/controller/AdminController.php [hunk 1]
```diff
@@ -436,38 +436,48 @@ public function __construct($forceControllerName = '', $default_theme_name = 'de
...
+            28 => $this->trans('A zone has been assigned to the selection successfully.', array(), 'Admin.Notifications.Success'),
+            29 => $this->trans('Successful upgrade', array(), 'Admin.Notifications.Success'),
+            30 => $this->trans('A partial refund was successfully created.', array(), 'Admin.OrdersCustomers.Notification'),
+            31 => $this->trans('The discount was successfully generated.', array(), 'Admin.Catalog.Notification'),
+            32 => $this->trans('Successfully signed in to PrestaShop Addons', array(), 'Admin.Modules.Notification'),
+        );
+
+        $this->_error = array(
+            1 => $this->trans(
+                'The root category of the shop %shop% is not associated with the current one.',
```
Comment (xBorderie) [inline classes/controller/AdminController.php]: The current one: current category or current shop?
Could you maybe clarify? :)

File: classes/controller/AdminController.php [hunk 2]
```diff
@@ -436,38 +436,48 @@ public function __construct($forceControllerName = '', $default_theme_name = 'de
...
-            24 => $this->l('Localization pack imported successfully.'),
-            25 => $this->l('The selected images have successfully been moved.'),
-            26 => $this->l('Your cover image selection has been saved.'),
-            27 => $this->l('The image\'s shop association has been modified.'),
-            28 => $this->l('A zone has been assigned to the selection successfully.'),
-            29 => $this->l('Successful upgrade'),
-            30 => $this->l('A partial refund was successfully created.'),
-            31 => $this->l('The discount was successfully generated.'),
-            32 => $this->l('Successfully signed in to PrestaShop Addons')
+            1 => $this->trans('Successful deletion', array(), 'Admin.Notifications.Success'),
```
Comment (xBorderie) [inline classes/controller/AdminController.php]: Some strings end with a full-stop, some don't. Could you make them all end with a "." for consistency? Thank you!

**Comment by xBorderie** [comment]
Thanks @aleeks for the wording changes! Still needs @AlexEven's expertise with domains though :)

File: classes/controller/AdminController.php [hunk 3]
```diff
@@ -436,38 +436,48 @@ public function __construct($forceControllerName = '', $default_theme_name = 'de
...
-            32 => $this->l('Successfully signed in to PrestaShop Addons')
+            1 => $this->trans('Successful deletion.', array(), 'Admin.Notifications.Success'),
+            2 => $this->trans('The selection has been successfully deleted.',  array(), 'Admin.Notifications.Success'),
+            3 => $this->trans('Successful creation.',  array(), 'Admin.Notifications.Success'),
+            4 => $this->trans('Successful update.',  array(), 'Admin.Notifications.Success'),
+            5 => $this->trans('The status has been successfully updated.', array(), 'Admin.Notifications.Success'),
+            6 => $this->trans('The settings have been successfully updated.', array(), 'Admin.Notifications.Success'),
+            7 => $this->trans('The image was successfully deleted.', array(), 'Admin.Notifications.Success'),
+            8 => $this->trans('The module was successfully downloaded.', array(), 'Admin.Modules.Notification'),
+            9 => $this->trans('The thumbnails were successfully regenerated.', array(), 'Admin.Notifications.Success'),
```
Comment (AlexEven) [inline classes/controller/AdminController.php]: Admin.Design.Notification?

File: classes/controller/AdminController.php [hunk 4]
```diff
@@ -436,38 +436,48 @@ public function __construct($forceControllerName = '', $default_theme_name = 'de
...
+            10 => $this->trans('The message was successfully sent to the customer.', array(), 'Admin.OrdersCustomers.Success'),
```
Comment (AlexEven) [inline classes/controller/AdminController.php]: Admin.OrdersCustomers.Notification

File: classes/controller/AdminController.php [hunk 5]
```diff
@@ -436,38 +436,48 @@ public function __construct($forceControllerName = '', $default_theme_name = 'de
...
+            15 => $this->trans('The translations have been successfully added.', array(), 'Admin.International.Notification'),
+            16 => $this->trans('The module transplanted successfully to the hook.', array(), 'Admin.Modules.Notification'),
+            17 => $this->trans('The module was successfully removed from the hook.', array(), 'Admin.Modules.Notification'),
+            18 => $this->trans('Successful upload.', array(), 'Admin.Notifications.Success'),
+            19 => $this->trans('Duplication was completed successfully.', array(), 'Admin.Notifications.Success'),
+            20 => $this->trans('The translation was added successfully, but the language has not been created.', array(), 'Admin.International.Notification'),
+            21 => $this->trans('Module reset successfully.', array(), 'Admin.Modules.Notification'),
+            22 => $this->trans('Module deleted successfully.', array(), 'Admin.Modules.Notification'),
+            23 => $this->trans('Localization pack imported successfully.', array(), 'Admin.International.Notification'),
+            24 => $this->trans('Localization pack imported successfully.', array(), 'Admin.International.Notification'),
```
Comment (AlexEven) [inline classes/controller/AdminController.php]: Why do we have it twice? (see line 23)

File: classes/controller/AdminController.php [hunk 6]
```diff
@@ -436,38 +436,48 @@ public function __construct($forceControllerName = '', $default_theme_name = 'de
...
+            32 => $this->trans('Successfully signed in to PrestaShop Addons.', array(), 'Admin.Modules.Notification'),
+        );
+
+        $this->_error = array(
+            1 => $this->trans(
+                'The root category of the shop %shop% is not associated with the current shop. You can\'t access this page. Please change the root category of the shop.',
+                array(
+                    '%shop%' => $this->context->shop->name,
+                ),
+                'Admin.Catalog.Feature'
```
Comment (AlexEven) [inline classes/controller/AdminController.php]: If this is indeed used in a catalog context, should be "Admin.Catalog.Notification"

File: classes/controller/AdminController.php (hunk 5 above)
Comment (aleeks) [inline classes/controller/AdminController.php]: Loul... here from long time

**Comment by dSkrbic** [comment]
Hello,

Test Q&A on public 1.7.0.4 and it's **OK**:

- When creating CatRoot -> no shop association asked
- When creating new shop -> association accepted with the CatRoot
- The implicite temporary association of the CatRoot with default shop is suppressed

So: OK, bravo Alex :)

**Comment by dSkrbic** [comment]
Removing the QA Approval because of incomplete test

**Comment by maximebiloe** [comment]
Thank you @aleeks!

**Pull request merged**